### PR TITLE
feat: Added delete events for dataEntry, dataMap and storyMap

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Added delete events for dataEntry, dataMap and storyMap

### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added

### CODE REVIEW

File: src/sharedData/visualization/components/VisualizationWrapper.js [hunk 1]
```diff
@@ -98,14 +100,17 @@ const VisualizationWrapper = props => {
...
   }, [imagePrinter, mapTitle]);
 
-  const onDelete = () => {
+  const onDelete = useCallback(() => {
     dispatch(deleteVisualizationConfig(data)).then(data => {
       const success = _.get('meta.requestStatus', data) === 'fulfilled';
       if (success) {
         onDeleted();
+        trackEvent('dataMap.delete', {
+          props: { owner: owner.slug, title: mapTitle },
```
Comment (paulschreiber) [inline src/sharedData/visualization/components/VisualizationWrapper.js:109]: question: do we need the map title? won't that identify the user? @DerekCaelin

File: src/sharedData/components/SharedDataEntryBase.js [hunk 2]
```diff
@@ -64,7 +64,9 @@ const SharedDataEntryBase = props => {
         const success = _.get('meta.requestStatus', data) === 'fulfilled';
         if (success) {
           updateOwner();
-          trackEvent('dataEntry.delete', { props: { owner: owner.slug } });
+          trackEvent('dataEntry.delete', {
+            props: { owner: owner.slug, name: dataEntry.name },
+          });
```
Comment (paulschreiber) [inline src/sharedData/components/SharedDataEntryBase.js:69]: question: do we need the name? won't that identify the user?

File: src/sharedData/components/SharedDataEntryBase.js [hunk 3]
```diff
@@ -84,7 +86,9 @@ const SharedDataEntryBase = props => {
         const success = _.get('meta.requestStatus', data) === 'fulfilled';
         if (success) {
           updateOwner();
-          trackEvent('dataEntry.edit', { props: { owner: owner.slug } });
+          trackEvent('dataEntry.edit', {
+            props: { owner: owner.slug, name: dataEntry.name },
+          });
```
Comment (paulschreiber) [inline src/sharedData/components/SharedDataEntryBase.js:91]: question: do we need the name? won't that identify the user?

**Comment by DerekCaelin** [comment]
@paulschreiber We can leave out the name of the deleted story map.

File: src/sharedData/visualization/components/VisualizationWrapper.js (hunk 1 above)
Comment (josebui) [reply src/sharedData/visualization/components/VisualizationWrapper.js:109]: I don't think that is user related data, it is the data map title, I think we are tracking the names of groups too in plausible

Comment (paulschreiber) [reply src/sharedData/visualization/components/VisualizationWrapper.js:109]: Groups have many users. A map is created by a single user. (Per Derek below, okay to omit.)

**Comment by DerekCaelin** [comment]
We discussed this in a team meeting.

[We have started a document](https://docs.google.com/document/d/1msvXBtOSCpX51y3FhoaxnW-LqrMJ17XORwTYitikS64/edit) to state when we collect user data and what constraints we're adopting. 

@paulschreiber @josebui please remove reference to the URL of the story map in the plausible events for story maps.

**Comment by josebui** [comment]
@DerekCaelin For this I would like more defined details, like:
```
Event: storymap.delete
Fields to track: title, deleted date
```

For the 3 events affected as part of this PR: storymap.delete, dataEntry.delete, dataMap.delete

Let me know if that is possible

**Comment by DerekCaelin** [comment]
Could we track the following?

**storymap.delete**
- date
- language of user

**dataEntry.delete**
- date
- group/landscape
- language of user
- name

**dataMap.delete**
- date
- group.landscape
- language of user
- name of map
- name of file

**Comment by paulschreiber** [comment]
Please upvote Plausible's request for built-in language tracking: https://github.com/plausible/analytics/discussions/865

@josebui some code snippets there might be useful.

**Comment by paulschreiber** [comment]
After discussing this again with others, I still believe including the file name makes it to easy to identify the user. If we need other information, such as how old the file was when it was deleted, we can capture that directly. (That's also more reliable than trying to manually correlate multiple events for the same object.)

@DerekCaelin If you want to discuss this further, let's have a quick call to sort it out.

**Comment by DerekCaelin** [comment]
@paulschreiber @josebui Can we please remove references to the **name** of files being shared or deleted, but **add** the duration of time on the site? I'm interested in knowing how long a file has existed on Terraso before being removed.

File: src/timeUtils.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+export const daysSince = dateString => {
+  const date = new Date(dateString);
+
+  if (isNaN(date)) {
+    return null;
+  }
+
+  const days = (new Date() - date) / (1000 * 60 * 60 * 24);
```
Comment (paulschreiber) [inline src/timeUtils.js:8]:
```suggestion
  const days = (new Date() - date) / (TIME_ONE_DAY);
```

File: src/timeUtils.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export const daysSince = dateString => {
```
Comment (paulschreiber) [inline src/timeUtils.js:1]:
```suggestion
const TIME_ONE_DAY = 86400000;
export const daysSince = dateString => {
```

File: src/timeUtils.js (hunk 4 above)
Comment (paulschreiber) [reply src/timeUtils.js:8]: How come this is 86,400,000 and not 86400? Are we using microseconds?

**Review by paulschreiber — APPROVED** [review]

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**